src/main/java/io/tus/java/client/ProtocolException.java

```diff
@@ -10,21 +10,38 @@
 public class ProtocolException extends Exception {
     private HttpURLConnection connection;
 
+    /**
+     * Instantiates a new Object of type {@link ProtocolException}.
+     * @param message Message to be thrown with the exception.
+     */
     public ProtocolException(String message) {
         super(message);
     }
 
+    /**
+     Instantiates a new Object of type {@link ProtocolException}.
+     * @param message Message to be thrown with the exception.
+     * @param connection {@link HttpURLConnection}, where the error occurred.
+     */
     public ProtocolException(String message, HttpURLConnection connection) {
         super(message);
         this.connection = connection;
     }
 
+    /**
+     * Returns the {@link HttpURLConnection} instances, which caused the error.
+     * @return {@link HttpURLConnection}
+     */
     public HttpURLConnection getCausingConnection() {
         return connection;
     }
 
+    /**
+     * Determines whether a retry attempt should be made after a {@link ProtocolException} or not.
```

Comment:
I could also tell something about response codes but it is also annotated below. I could also merge it

Reply:
Don´t worry, that's good enough!